wordpress-mobile/WordPress-iOS
# Reader post comments following: refactor ObjC methods to Swift

Ref: #17632 
Depends on: #17717

This adds a `ReaderCommentsFollowHelper` to (eventually) contain all the code necessary for the Follow Conversation flows. Right now, it contains:
- A Swift version of the `ReaderCommentsViewController.m` method `handleFollowConversationButtonTapped`.
- A Swift version of the `ReaderCommentsViewController.m` method `handleNotificationsButtonTappedWithUndo`.

Notes:
- This does not move _all_ the Follow Conversation related code to `ReaderCommentsFollowHelper`. There is a bit from `ReaderCommentsViewController.swift`, and `ReaderCommentsNotificationSheet` handling. To limit this PR though, I've only handled the ObjC methods for now. 
- This refactor doesn't really need to be dependent on the `followConversationPostDetails` feature flag. However, I didn't want to make such sweeping changes this close to code freeze unnecessarily. I'll remove this dependency for the next release when the refactor is finished and we have more time to fully test functionality.

Also, I'm not set on the name `ReaderCommentsFollowHelper`. If you have any suggestions lemme know!

To test:
- Enable the `followConversationPostDetails` feature flag.
- Go to Reader, and select any unsubscribed post.
- 🔍 Verify that the "Follow" button should be visible on the navigation bar.
- Tap the Follow button.
- 🔍 Verify that the snackbar with the "Enable" action button is shown.
- Tap on the "Enable" button.
- 🔍 Verify that a feedback notice with an "Undo" action button is shown.
- Tap on the "Undo" action button.
- 🔍 Verify that a feedback notice is shown.
- 🔍 Verify that the Follow button has turned into a bell icon button in the navigation bar.
- Go to Reader, and select any subscribed post.
- Tap on the bell icon button.
- 🔍 Verify that the bottom sheet is shown.
- Tap on the switch.
- Dismiss the bottom sheet.
- 🔍 Verify that a feedback notification is shown, and correctly represents the desired state.
- Tap the bell icon button again.
- Tap on the Unfollow Conversation button.
- 🔍 Verify that a feedback notification is shown.
- 🔍 Verify that the bell icon button has now turned back into "Follow".

(@dvdchr yes, I totally stole these testing steps from your [PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/17714). Thanks for the help! 😆 )

## Regression Notes
1. Potential unintended areas of impact
N/A. Feature is disabled.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature is disabled.

3. What automated tests I added (or what prevented me from doing so)
N/A. Feature is disabled.

PR submission checklist:
